persistence/modeshape-persistence-relational/src/main/java/org/modeshape/persistence/relational/StatementsProvider.java

```diff
@@ -104,6 +108,26 @@ protected Void dropTable(Connection connection) throws SQLException {
         }
     }
 
+    protected <T> T load(Connection connection, Set<String> ids, Function<ResultSet, T> function) throws SQLException {
+        logDebug("Loading ids {0} from {1}", ids.toString(), tableName);
+        String params = ids.stream().map(id -> "?").collect(Collectors.joining(","));
+        String statementString = statements.get(GET_MULTIPLE).replaceAll("#", params);
+        try (PreparedStatement ps = connection.prepareStatement(statementString)) {
+            AtomicInteger counter = new AtomicInteger(1);
+            ids.forEach(id -> {
+                try {
+                    ps.setString(counter.getAndIncrement(), id);
+                } catch (SQLException e) {
+                    throw new RelationalProviderException(e);
+                }
+            });
+            try (ResultSet rs = ps.executeQuery()) {
+                return function.apply(rs);
+            }
+        }
+    }
+    
+
```

Comment:
Is there a limit to the number of IDs that a single statement can handle? Should this method perhaps do no more than X identifiers at a time?